src/eventshub/src/component/Autocomplete/Autocomplete.tsx

```diff
@@ -28,152 +20,224 @@ type AutocompleteProps = Partial<AutocompleteOptions<AlgoliaVenueItem>> & {
-            {hit.space_name && hit.space_name !== hit.venue_name && (
-              <span> - {hit.space_name}</span>
-            )}
-          </div>
-        </div>
-      </div>
-    </div>
-  );
-};
 /**
  * Autocomplete component.
  */
 export function Autocomplete({
   searchClient,
   searchIndex,
-  className,
   activeTab,
   activeField,
-  ...autocompleteProps
+  placeholder,
+  openOnFocus,
 }: AutocompleteProps) {
-  const containerRef = useRef<HTMLDivElement | null>(null);
-  const panelRef = useRef<HTMLDivElement | null>(null);
-  const panelRootRef = useRef<Root | null>(null);
-  const rootRef = useRef<HTMLElement | null>(null);
-
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [autocompleteState, setAutocompleteState] = useState<AutocompleteState<AlgoliaVenueItem>>({
+    query: '',
+    collections: [],
+    isOpen: false,
+    activeItemId: null,
+    completion: null,
+    context: {},
+    status: 'idle',
+  });
   const { refine: setQuery, query: activeQuery } = useSearchBox();
   const { refine: setPage } = usePagination();
   const { filters, getRefinedFilters } = useFilters();
   const { combined } = useCombinedRefinement();
-  const suggestionsIndex = algoliaSuggestionsIndexMap[activeTab];
 
   const getFilterQuery = useCallback(() => {
     const refined = getRefinedFilters(combined, filters, activeTab);
     const inlined = refined.map((item) => `${activeField}:${item}`);
     return [inlined]; // Wrapped in an array to be used as an or filter.
   }, [combined, filters, activeTab, activeField, getRefinedFilters]);
 
-  useEffect(() => {
-    if (!containerRef.current || !panelRef.current) {
-      return;
-    }
 
-    const autocompleteInstance = autocomplete({
-      container: containerRef.current,
-      // panelContainer: panelRef.current,
-      insights: true,
-      id: nanoid(),
-      getSources({ query }) {
-        return [
-          {
-            sourceId: activeTab,
-            getItemInputValue: ({ item }) => {
-              let ret = item.name;
-              if (item.space_name && item.space_name !== item.venue_name) {
-                ret = `${item.venue_name} - ${item.space_name}`;
-              }
-              return ret;
-            },
-            getItems() {
-              return getAlgoliaResults({
-                searchClient,
-                queries: [
-                  {
-                    indexName: searchIndex,
-                    query,
-                    params: {
-                      facetFilters: getFilterQuery(),
-                    },
-                  },
-                ],
-              });
-            },
-            templates: {
-              item({ item, components }) {
-                return <SuggestionHit hit={item} components={components} />;
+  const autocomplete = useMemo(
+    () =>
+      createAutocomplete<AlgoliaVenueItem, CombinedEvent, ReactMouseEvent, ReactKeyBoardEvent>({
```

Comment:
New function to allow us to create our own 'headless' autocomplete. This also should allow us to skip any issues with the different react version requirements/compatibility with instantsearch.